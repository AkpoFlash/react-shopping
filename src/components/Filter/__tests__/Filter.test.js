import React from 'react';
import { Provider } from 'react-redux';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import ConnectedFilter, { Filter } from '../Filter';

describe('Dumb Filter', () => {
	const shallowWrapper = shallow(<Filter />);
	const mountWrapper = mount(<Filter />);

	it('check Filter shapshot', () => {
		expect(shallowWrapper).toMatchSnapshot();
	});

	it('Filter must be the list', () => {
		expect(mountWrapper.find('ul').exists()).toBe(true);
	});

	it('Filter contain element with "all" name', () => {
		expect(mountWrapper.find('ul li[name="all"]').exists()).toBe(true);
	});

});

// Enzyme is not supported new `memo` feature from React 16.6, without `memo` all work good

// describe('Connected Filter', () => {
// 	const mockStore = configureStore();
// 	const initialState = {
// 		filter: {
// 			filterBy: 'all',
// 			searchQuery: 'Test',
// 		},
// 		languages: {
// 			userLang: 'Rus',
// 		},
// 	};
// 	const store = mockStore(initialState);
// 	const mountWrapper = mount(<Provider store={store}><ConnectedFilter /></Provider>);

// 	it('Filter\'s prop searchQuery equalens InitialState', () => {
// 		expect(mountWrapper.find('Filter').prop('searchQuery')).toBe(initialState.filter.searchQuery);
// 	});

// 	it('Filter\'s search input must be equalens InitialState', () => {
// 		expect(mountWrapper.find('Filter input')).toHaveValue(initialState.filter.searchQuery);
// 	});

// });