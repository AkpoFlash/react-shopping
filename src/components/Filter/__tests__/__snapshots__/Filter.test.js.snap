// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dumb Filter check Filter shapshot 1`] = `
<styled.ul>
  <styled.li
    data-active={false}
    onClick={[Function]}
  >
    All
  </styled.li>
  <styled.li
    data-active={false}
    onClick={[Function]}
  >
    Popular
  </styled.li>
  <styled.li
    data-active={false}
    onClick={[Function]}
  >
    Price (Expensive)
  </styled.li>
  <styled.li
    data-active={false}
    onClick={[Function]}
  >
    Price (Cheap)
  </styled.li>
  <styled.li
    data-active={false}
    onClick={[Function]}
  >
    Author
  </styled.li>
  <styled.input
    name="search"
    onChange={[Function]}
    placeholder="Search..."
  />
</styled.ul>
`;
